cpp/cmake_modules/ThirdpartyToolchain.cmake

```diff
@@ -820,6 +820,8 @@ if (ARROW_WITH_RAPIDJSON)
     ExternalProject_Add(flatbuffers_ep
       URL ${FLATBUFFERS_SOURCE_URL}
       CMAKE_ARGS
+      "-DCMAKE_C_COMPILER=${CMAKE_C_COMPILER}"
```

Comment:
Don't add the compilers but use `EP_COMMON_CMAKE_ARGS` instead. This will also add AR, RANLIB and CFLAGS

Reply:
The problem is that the build type has to be overridden. I'll split the common args into a variable that contains the compiler toolchain stuff